src/core/identifiers/identifiers.ts

```diff
@@ -18,7 +19,7 @@ export type DSNPAnnouncementURI = string;
  */
 export const isDSNPAnnouncementURI = (id: unknown): id is DSNPAnnouncementURI => {
   if (!isString(id)) return false;
-  return id.match(/^dsnp:\/\/0x[0-9A-F]{1,16}\/0x[0-9A-F]{64}$/i) !== null;
+  return id.match(/^dsnp:\/\/0x[1-9a-f][0-9a-f]{0,15}\/0x[1-9a-f][0-9a-f]{0,63}$/) !== null;
```

Comment:
we did we drop case-insensitivity here? are we requiring that hex be lowercase too?